vm/systemSmartContracts/esdt.go

```diff
@@ -1225,6 +1254,33 @@ func (e *esdt) setSpecialRole(args *vmcommon.ContractCallInput) vmcommon.ReturnC
 	return vmcommon.Ok
 }
 
+func checkCorrectAddressForNFTCreateMultiShard(token *ESDTDataV2) error {
+	nftCreateRole := []byte(core.ESDTRoleNFTCreate)
+	mapLastBytes := make(map[uint8]struct{})
+	for _, esdtRole := range token.SpecialRoles {
+		hasCreateRole := false
+		for _, role := range esdtRole.Roles {
+			if bytes.Equal(nftCreateRole, role) {
+				hasCreateRole = true
+				break
+			}
+		}
+
+		if !hasCreateRole {
+			continue
+		}
+
+		lastBytesOfAddress := esdtRole.Address[len(esdtRole.Address)-1]
+		if _, exist := mapLastBytes[lastBytesOfAddress]; exist {
```

Comment:
"exists" and please split on multiple lines

Reply:
done